2013-04-01  Magnus Granberg <zorry@gentoo.org>

	* gcc/config/arm/arm.h (SUBTARGET_EXTRA_SPECS): Add ESP_EXTRA_SPECS.
	(DRIVER_SELF_SPECS): Add ESP_DRIVER_SELF_SPEC.

--- a/gcc/config/arm/arm.h	2013-01-15 17:17:28.000000000 +0100
+++ b/gcc/config/arm/arm.h	2013-02-18 22:45:18.327284928 +0100
@@ -226,8 +226,12 @@ extern void (*arm_lang_output_object_att
   SUBTARGET_EXTRA_SPECS
 
 #ifndef SUBTARGET_EXTRA_SPECS
+#ifdef ENABLE_ESP
+#define SUBTARGET_EXTRA_SPECS ESP_EXTRA_SPECS
+#else
 #define SUBTARGET_EXTRA_SPECS
 #endif
+#endif
 
 #ifndef SUBTARGET_CPP_SPEC
 #define SUBTARGET_CPP_SPEC      ""
@@ -2326,6 +2326,11 @@ extern const char *host_detect_local_cpu
 # define MCPU_MTUNE_NATIVE_SPECS ""
 #endif
 
-#define DRIVER_SELF_SPECS MCPU_MTUNE_NATIVE_SPECS
-
+#ifdef ENABLE_ESP
+# define DRIVER_SELF_SPECS \
+  MCPU_MTUNE_NATIVE_SPECS, \
+  ESP_DRIVER_SELF_SPEC
+#else
+# define DRIVER_SELF_SPECS MCPU_MTUNE_NATIVE_SPECS
+#endif
 #endif /* ! GCC_ARM_H */
